app/models/local_authority.rb

```diff
@@ -14,6 +14,13 @@ def provided_services
     Service.for_tier(self.tier).enabled
   end
 
+  def calculate_count_of_broken_links
+    # to ensure that we can replay our migrations, wrap the update in a guard clause.
+    if respond_to?(:broken_link_count)
+      update_attribute(:broken_link_count, self.links.where.not(status: 200).count)
```

Comment:
I really like that. It makes complete sense that when we talk about Broken Links that they be either a subset of all links aka. `Link.all.broken` as you suggest or even a first class model object eg. `class BrokenLink < Link`